src/usr/local/www/system.php

```diff
@@ -139,11 +139,11 @@ function is_timezone($elt) {
 	    sprintf(ngettext('(%1$s hour %2$s GMT)', '(%1$s hours %2$s GMT)', intval($hr_offset)), $hr_offset, $direction_str);
 }
 
-$multiwan = false;
+$multiwan = 0;
 $interfaces = get_configured_interface_list();
 foreach ($interfaces as $interface) {
 	if (interface_has_gateway($interface)) {
-		$multiwan = true;
+		$multiwan++;
```

Comment:
Why not break foreach here instead of changing variable to int?

Reply:
That's how it was before, but you're right it's not efficient. I just pushed a small change to break out of the loop as soon as its >1.

I wonder if the whole "hide the gateways menu if there is only 1" is even worth it. There doesn't seem to be much downside to me of always showing the popup menus, and they will just default to "none" on single-WAN systems.

It could even be useful—I wonder what happens when you have a gateway selected for a DNS server and then that gateway is removed. Does the DNS change back to "none"? Or does the system prevent you from deleting it? If not, then the config might end up with a DNS server bound to a gateway and since the GUI is now hidden, no way to change it back aside from deleting/re-adding (I haven't tested this).